neo/app/actor/RequestRouter.scala

```diff
@@ -0,0 +1,107 @@
+package actor
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import akka.pattern.{ask, pipe}
+import akka.stream.Materializer
+import akka.util.Timeout
+import edu.uci.ics.cloudberry.zion.common.Config
+import play.api.libs.json._
+import play.api.libs.ws.WSClient
+import play.api.mvc.RequestHeader
+import play.Logger
+
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext
+
+class RequestRouter (out: ActorRef, ws: WSClient, requestHeader: RequestHeader, berryClientProp: Props, config: Config)
+                    (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
+
+  import RequestRouter._
+  import KeyType._
+
+  implicit val timeout: Timeout = Timeout(20.minutes)
+  val StreamingBerryClient = context.actorOf(berryClientProp)
+  val nonStreamingBerryClient = context.actorOf(berryClientProp)
+
+  // TODO UserRequest in package model is of no use
+  override def receive: Receive = {
+    case requestBody: JsValue =>
+      (requestBody \ "transform" \\ "key")(0).asOpt[KeyType.Value] match {
+        case Some(TotalCount) =>
```

Comment:
I think `TotalCount` and the `Sample` is the value of the `key` field. As a general purpose request handler, it shouldn't handle the user's logic.

I think my previous idea of simplification in Slack is not precise. My bad 😭 

The correct way should be something like following
```
 case body : JsValue =>
    val transform = parseTransform(body)
    (body \\ "slicing") match {
          case Some(x) => handleStreaming(x, transform)
          case None => handleNonStreaming(body, transform)
     }
...

// here is the simplification I meant yesterday.
def parseTransform(body: JsValue) = {
   (body \\ "wrapper").asOpt[JsValue] match {
        case None => Transform.NonTransform
        case Some(x) => Transform.Wrapper( (x \ "key").as[String], (x \ "value").as[String]) ... 
   } 
}

trait Transform = {
   def tranform(jsvalue: JsValue): JsValue
}
object Transfrom {
  val NoTransform = new Transform ...
  val WraperTransform(key: String, value: String) = new Transform ....
}
```